clouds/azure/utils.py

```diff
@@ -0,0 +1,31 @@
+from typing import List
+from simple_logger.logger import get_logger
+from azure.mgmt.subscription import SubscriptionClient
+from azure.mgmt.resource import ResourceManagementClient
+from azure.mgmt.redhatopenshift import AzureRedHatOpenShiftClient
+
+
+LOGGER = get_logger(name=__name__)
+
+
+def get_aro_supported_versions(aro_client: AzureRedHatOpenShiftClient, region: str) -> List[str]:
+    supported_versions = [aro_version.version for aro_version in aro_client.open_shift_versions.list(location=region)]
+    LOGGER.info(f"ARO supported versions: {supported_versions}")
+    return supported_versions
+
+
+def get_azure_supported_regions(subscription_client: SubscriptionClient, subscription_id: str) -> List[str]:
+    supported_regions = [
+        region.name for region in subscription_client.subscriptions.list_locations(subscription_id=subscription_id)
+    ]
+    LOGGER.info(f"ARO supported regions: {supported_regions}")
+    return supported_regions
+
+
+def azure_cleanup(resource_client: ResourceManagementClient) -> None:
```

Comment:
1. please rename to be more specific
2. please add docsting
3. with the current code, the function exists on first failure. does `begin_delete` raise anything? does it make sense to continue after a failure to next resources? does `begin_delete` report failure in another way?
4. with `begin_delete` - do we need to wait for deletion or will the function return after deletion?

Reply:
1. Done
2. Done, let me know if thats ok
3. The `begin_delete` raises a `azure.core.exceptions.HttpResponseError` exception, which will report the failure message with the inherited appropriate exception (e.g. `azure.core.exceptions.ResourceNotFoundError`).
I found it reasonable to continue after any delete failure while reporting them during/post cleanup.
Let me know what you think.
4. We dont need to wait, the function will continue to delete the next resource group only when the current is destroyed.